plugin/src/main/java/org/opensearch/sql/plugin/SQLPlugin.java

```diff
@@ -195,12 +223,49 @@ public ScriptEngine getScriptEngine(Settings settings, Collection<ScriptContext<
 
   @Override
   public void reload(Settings settings) {
-    CatalogServiceImpl.getInstance().loadConnectors(settings);
-    CatalogServiceImpl.getInstance().registerDefaultOpenSearchCatalog(openSearchStorageEngine());
+    dataSourceService.clear();
+    dataSourceService.addDataSource(defaultOpenSearchDataSourceMetadata());
+    loadDataSources(dataSourceService, settings);
   }
 
-  private StorageEngine openSearchStorageEngine() {
-    return new OpenSearchStorageEngine(new OpenSearchNodeClient(client), pluginSettings);
+  /**
+   * load {@link DataSource} from settings.
+   */
+  @VisibleForTesting
+  public static void loadDataSources(DataSourceService dataSourceService, Settings settings) {
+    SecurityAccess.doPrivileged(
+        () -> {
+          InputStream inputStream = DataSourceSettings.CATALOG_CONFIG.get(settings);
+          if (inputStream != null) {
+            ObjectMapper objectMapper = new ObjectMapper();
+            objectMapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+            try {
+              List<DataSourceMetadata> metadataList =
+                  objectMapper.readValue(inputStream, new TypeReference<>() {});
+              verifyDuplicateName(metadataList);
```

Comment:
Now the interfaces are much more clearer. Any reason, we are adding single data source. can we expose one more method for a list and take the duplicate name validation to DatasourceServiceImpl class itself.

Reply:
Done. Refactor the addDataSource() and add validation.